inginious/frontend/pages/course_admin/student_list.py

```diff
@@ -284,6 +292,35 @@ def post_groups(self, course, data, active_tab, msg, error):
                 msg["groups"] = _('An error occurred while parsing the data.')
                 error["groups"] = True
             active_tab = "tab_groups"
+        if "audiencecreationfile" in data and 'upload_audiences_creation' in data:
+            self.database.audiences.delete_many({"courseid": course.get_id()})
+            text_stream = io.TextIOWrapper(data["audiencecreationfile"], encoding='utf-8')
+            reader = csv.reader(text_stream)
+            stud_list, _, _, _ = self.get_user_lists(course)
+
+            def _list_checker(users):
+                inserted_content = []
+                for usr in users:
+                    regex = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'
+                    if re.fullmatch(regex, usr):
+                        user = self.database.users.find_one({"email": usr})
+                        user is not None and inserted_content.append(user["username"])
+                    else:
+                        inserted_content.append(usr)
+                return inserted_content
+
+            for row in reader:
+                stud = row[1].split(';') if len(row) > 1 else []
```

Comment:
CSV files can be seen as relational tables. I suppose here you consider the file format such as there is one line per audience. The additional value compared to a "database-dump" YAML document will be limited as this is rarely the way spreadsheets are produced. 

Administrators often start from the student list. Therefore, a handy format would be something like this:

    user_id | user_role | audience_description

where ``user_id`` is by default, the email address, which is a commonly available piece of information, ``user_role``, a numerical or textual enumeration defining if the user is a student or a tutor, and ``audience_description`` the description of the new audience, which is considered to be unique per audience at the import time (it would, by the way, actually be accurate to also make this unique in database).

A further improvement would be to allow user identification based on additional fields. For instance:

    user_id | id_field | user_role | audience_description
    
where id_field could be the DB field we want to match (``username``, ` `email``, ``bindings.auth_id/field``). A binding-based match would however require more control at the course level as this could violate GDPR, but the two first methods can be implemented.